addons/mail/controllers/discuss/gif.py

```diff
@@ -0,0 +1,58 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import json
+import requests
+from odoo.http import request, route, Controller
+
+
+class DiscussGifController(Controller):
+
+    def _request_gifs(self, url):
```

Comment:
While I understand that you think it could make the code better, I think it would be a mistake for the following reasons:

* Each route target an different endpoint of the tenor API, so my dict will have to contains the endpoint.
* Each endpoints have some dedicated query params that are not used by the others. In an ideal world I would check that we correctly use the right params for the right endpoints.

That will make it harder to understand who use what and how.
We currently use 3 endpoints of the Tenor API, I don't think we need an url crafting method.

Reply:
however, a good way to avoid abuse is to expect an endpoint and not a full URL, and let this method add the tenor base url